pennylane/spin/lattice.py

```diff
@@ -71,12 +71,84 @@ class Lattice:
 
     **Example**
 
-    >>> n_cells = [2, 2]
-    >>> vectors = [[0, 1], [1, 0]]
-    >>> boundary_condition = [True, False]
-    >>> lattice = qml.spin.Lattice(n_cells, vectors, boundary_condition=boundary_condition)
+    We can define the positions of nodes in the lattice unit cell along with the lattice vectors
+    to create a custom lattice layout.
+
+    .. code-block:: python
+
+        from pennylane.spin import Lattice
+
+        positions = [[0.2, 0.5],
+                     [0.5, 0.2],
+                     [0.5, 0.8],
+                     [0.8, 0.5]]
+
+        vectors = [[1, 0], [0, 1]]
+
+        n_cells = [2, 2]
+
+        # periodic boundadry conditions apply along the [1,0] axis only
+        boundary_condition = [True, False]
+
+        lattice = Lattice(n_cells, vectors, positions, boundary_condition=boundary_condition)
+
     >>> lattice.edges
-    [(2, 3, 0), (0, 2, 0), (1, 3, 0), (0, 1, 0)]
+    [(10, 13, 0), (0, 11, 0), (4, 15, 0), (2, 5, 0), (3, 8, 0), (7, 12, 0)]
```

Comment:
Not sure what's happening but for this I'm getting the following output in the RC branch on a jupyter notebook
```python
[(np.int64(10), np.int64(13), 0),
 (np.int64(0), np.int64(11), 0),
 (np.int64(4), np.int64(15), 0),
 (np.int64(2), np.int64(5), 0),
 (np.int64(3), np.int64(8), 0),
 (np.int64(7), np.int64(12), 0)]
```

It seems to be a numpy 2.0 issue. I ran the same code in another env with numpy 1.26 and got the expected output. I guess it's okay since we can't control which version of numpy users use? But it might also make sense for our examples to be consistent with the latest version of numpy we support. Not sure exactly how to proceed :(

Reply:
RC branch supports numpy 2.0?

Reply:
Yep

Reply:
Will be interesting to know if this just an issue with this feature or there are other examples that print ints this way.

Reply:
discussed offline and decided to leave the docstring showing the simpler output